### PR TITLE
 🐛 Remove lint-full dependency

### LINKED ISSUE #2286: Running lint-full twice in CI
**What steps did you take and what happened:**
`make lint-full` needs to only be run once in CI, not twice.

[This](https://github.com/kubernetes-sigs/cluster-api/blob/master/Makefile#L473) gets called from [this](https://github.com/kubernetes-sigs/cluster-api/blob/master/scripts/ci-integration.sh#L66)

And [this](https://github.com/kubernetes-sigs/cluster-api/blob/master/scripts/ci-make.sh#L23) is a ci call to lint-full. 

Do we really need both of these to be run on both CI jobs?

/kind bug
/milestone Next
/priority backlog
/help

### PR DESCRIPTION
**Pull request by prankul88**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2286

### CODE REVIEW

**Comment by detiber** [comment]
I believe the integration test failure is due to the removal of the `generate` target as a prerequisite...

**Comment by vincepri** [comment]
I'd add `generate` to ci script instead of a pre-req in the Makefile

**Comment by chuckha** [comment]
> I'd add `generate` to ci script instead of a pre-req in the Makefile

To expand a little bit, the reason this makes sense is that the only thing calling docker-build-example-provider is CI and it's better to be explicit about what exactly CI needs to run.

The downside of this approach is that if someone is running this locally, they'd have to remember to run `generate` if they modified the API, **but** I don't think this is a workflow that is used outside of CI anyway, so it would be better to make it explicit in the ci script than in the Makefile.

**Comment by prankul88** [comment]
@detiber @fabriziopandini @vincepri @chuckha I have added `generate` to `ci-integration.sh`. PTAL

**Comment by chuckha** [comment]
/lgtm

Thank you for addressing the feedback!

File: scripts/ci-integration.sh [hunk 1]
```diff
@@ -33,6 +33,8 @@ CONTROLLER_REPO_KUBEADM_BOOTSTRAP="controller-ci-kubeadm-bootstrap"
 CONTROLLER_REPO_KUBEADM_CONTROL_PLANE="controller-ci-kubeadm-control-plane"
 EXAMPLE_PROVIDER_REPO="example-provider-ci"
 CERT_MANAGER_URL="https://github.com/jetstack/cert-manager/releases/download/v0.11.0/cert-manager.yaml"
+TOOLS_DIR="hack/tools"
+TOOLS_BIN_DIR="${TOOLS_DIR}/bin"
```
Comment (detiber) [inline scripts/ci-integration.sh]: Are these definitions needed? They seem to replicate the defaults already defined in the Makefile.

File: scripts/ci-integration.sh [hunk 2]
```diff
@@ -57,12 +59,19 @@ install_kustomize() {
 
 build_containers() {
    VERSION="$(git describe --exact-match 2> /dev/null || git describe --match="$(git rev-parse --short=8 HEAD)" --always --dirty --abbrev=8)"
+   export CONTROLLER_GEN="${PWD}/${TOOLS_BIN_DIR}/controller-gen"
+   export CONVERSION_GEN="${PWD}/${TOOLS_BIN_DIR}/conversion-gen"
+   export KUSTOMIZE="${PWD}/${TOOLS_BIN_DIR}/kustomize"
+   export GOBINDATA="${PWD}/${TOOLS_BIN_DIR}/go-bindata"
+   export GOBINDATA_CLUSTERCTL_DIR="cmd/clusterctl/config"
+   export CERTMANAGER_COMPONENTS_GENERATED_FILE="cert-manager.yaml"
```
Comment (detiber) [inline scripts/ci-integration.sh]: I don't think we need these defined here, they seem to be replicating the defaults that already exist in the Makefile

Comment (prankul88) [reply scripts/ci-integration.sh]: @detiber Sorry I thought maybe it is required to declare these variables locally to the file.

Comment (chuckha) [reply scripts/ci-integration.sh]: my fault for letting it get past my review, i'll file an issue to look into this

Comment (detiber) [reply scripts/ci-integration.sh]: No worries, @prankul88 has already filed a followup PR.

**Pull request merged**